coalahtml/_coalahtml/app/controllers/main.js

```diff
@@ -1,41 +1,32 @@
 'use strict';
 
 angular.module('coalaHtmlApp')
-  .controller('MainCtrl', function ($scope, $http) {
-    var parseCoalaProject = function(projectDir, coalaJSON) {
-      $http.get(projectDir + $scope.Constants.file_data)
-        .success(function(file_data) {
-          $scope.file_dict = angular.copy(file_data);
-          coalaJSON = coalaJSON || projectDir + $scope.Constants.coala;
-          $scope.data = [];
-          $http.get(coalaJSON).success(function(coala_json) {
-            var knownFiles = {};
-            var parseResult = function(el) {
-                el.affected_code.forEach(function(sourceRange){
-                    if (!(sourceRange.file in knownFiles)){
-                        knownFiles[sourceRange.file] = [];
-                    }
-                    knownFiles[sourceRange.file].push({
-                      "start":    sourceRange.start.line,
-                      "end":      sourceRange.end.line,
-                      "diffs":    el.diffs,
-                      "message":  el.message,
-                      "origin":   el.origin,
-                      "severity": el.severity
-                    });
-                });
-            };
-            for (var section in coala_json.results) {
-              if (coala_json.results[section].length > 0) {
-                coala_json.results[section].forEach(parseResult);
-              }
+  .controller('MainCtrl', function ($scope, $rootScope) {
+    var parseCoalaProject = function() {
+        $scope.data = [];
+        var knownFiles = {};
+        var parseResult = function(el) {
+          el.affected_code.forEach(function(sourceRange) {
+            if (!(sourceRange.file in knownFiles)) {
+              knownFiles[sourceRange.file] = [];
             }
-            $scope.data.push(knownFiles);
+            knownFiles[sourceRange.file].push({
+              "start":    sourceRange.start.line,
+              "end":      sourceRange.end.line,
+              "diffs":    el.diffs,
+              "message":  el.message,
+              "origin":   el.origin,
+              "severity": el.severity
+            });
           });
-        });
+        };
+        for (var section in $rootScope.COALA_JSON.results) {
+          if ($rootScope.COALA_JSON.results[section].length > 0) {
+            $rootScope.COALA_JSON.results[section].forEach(parseResult);
+          }
+        }
+        $scope.data.push(knownFiles);
       };
-    $http.get("data/Constants.json").success(function(constants) {
-      $scope.Constants = constants;
-      parseCoalaProject($scope.Constants.data);
-    });
+
+    parseCoalaProject();
```

Comment:
I don't think it's very probable that you'll want to parse coala project more than once, but if you'll more functions here, this way it'll be easier to read, so I think it can stay as it is.